### PR TITLE
Release 2022.7.1

### PR DESCRIPTION
**Pull request by jrbourbeau**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jrbourbeau** [comment]
@jakirkham we're seeing 

```python
RuntimeError: Could not download https://pypi.io/packages/source/d/dask/dask-2022.7.1.tar.gz
```

in CI. Do you know if this download error has popped up in other projects recently?

**Comment by jakirkham** [comment]
No clue. Was able to download from that link just now. Maybe a hiccup?

@conda-forge-admin, please restart CI

**Comment by jakirkham** [comment]
Interesting. On closer inspection it seem the file it is downloading is empty. Not seeing this with the previous release. Would it be possible to reupload the `sdist`?

**Comment by jakirkham** [comment]
Should add tried to download an `sdist` from a project that was uploaded a couple mins ago and didn't have this issue. In particular this one

```bash
curl -LO https://pypi.io/packages/source/t/tooltime/tooltime-0.2.4.tar.gz
```

Guessing something didn't work correctly when the `dask` `sdist` was uploaded (maybe it got corrupted on PyPI somehow)?

**Comment by jrbourbeau** [comment]
> On closer inspection it seem the file it is downloading is empty.

Hrm, interesting indeed. When I go to PyPI and download `dask-2022.7.1.tar.gz` I see a directory with contents. 

> Would it be possible to reupload the sdist?

I'm not sure how to do this. Does it mean rereleasing somehow on PyPI?

**Comment by jakirkham** [comment]
Yeah there are a number of options. One simple one might be just to delete the `sdist` and upload it again. Here's the relevant UI

Screen Shot 2022-07-22 at 2 45 09 PM

**Comment by jrbourbeau** [comment]
What about the "You will not be able to re-upload a new distribution of the same type with the same version number." disclaimer in that warning message? Naively reading that makes me think I may not be able to re-upload [dask-2022.7.1.tar.gz](https://files.pythonhosted.org/packages/68/e1/1edb5eeb15f88267075c37ee8c83c4527e462bdea328566c5b1e88db705c/dask-2022.7.1.tar.gz)

**Comment by jakirkham** [comment]
That's unfortunate. I suppose another option would be to delete the release and reupload?

Screen Shot 2022-07-22 at 2 51 07 PM

**Comment by jrbourbeau** [comment]
Stepping back a bit, I still don't understand why we're getting an empty `sdist` from https://pypi.io in CI. When I go to PyPI directly the `sdist` for 2022.7.1 isn't empty. Do you know how pypi.io and pypi.org relate to one another?

**Comment by jakirkham** [comment]
My understanding is it suppose to be a redirect. However there may be more going on behind the scenes.

If we don't want to go with the deletion route, would suggest raising an issue on this repo ( https://github.com/pypi/warehouse ). It seems there has been at least one other issue like this in the past ( for example https://github.com/pypi/warehouse/issues/7070 ). Though I don't recall seeing something like this before.

**Comment by jrbourbeau** [comment]
Yeah, I'm in a bit nervous about the deletion route. My understanding is we generally can't re-upload releases (though I may be wrong about that) 

Thanks for pointing me to `pypi/warehouse`. I went and opened up this issue https://github.com/pypi/warehouse/issues/11934

**Comment by jakirkham** [comment]
Should add we can use the URL with hashes in it for the interim (while that upstream issue gets worked through). They are a bit suboptimal as they need regenerating each update, but could be a short-term solution.

**Comment by jrbourbeau** [comment]
I don't have any objection to temporarily using the pypi.org URL with the hash. I do have to step away from the keyboard for an hour-ish. If you're got bandwidth, feel free to push to this branch. But no worries if you're busy with other things, I can always update things here when I'm back online

File: recipe/meta.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ package:
 
 source:
   url: https://pypi.io/packages/source/d/dask/dask-{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipe/meta.yaml:9]: Suggesting temporarily using the hashed link while the upstream issue with the redirect one ( https://github.com/pypi/warehouse/issues/11934 ) gets fixed

```suggestion
  url: https://files.pythonhosted.org/packages/68/e1/1edb5eeb15f88267075c37ee8c83c4527e462bdea328566c5b1e88db705c/dask-2022.7.1.tar.gz
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
 
 package:
   name: dask-core
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/d/dask/dask-{{ version }}.tar.gz
-  sha256: 50ad63035d6429cfa71d4c1832888a668e4260b1ef11163c795edc0ed177b4b3
+  url: https://files.pythonhosted.org/packages/68/e1/1edb5eeb15f88267075c37ee8c83c4527e462bdea328566c5b1e88db705c/dask-2022.7.1.tar.gz
```
Comment (jakirkham) [inline recipe/meta.yaml:9]: Template `version` here to make sure this URL gets fixed during the next release

```suggestion
  url: https://files.pythonhosted.org/packages/68/e1/1edb5eeb15f88267075c37ee8c83c4527e462bdea328566c5b1e88db705c/dask-{{ version }}.tar.gz
```

**Comment by jakirkham** [comment]
No worries. It is simple to add

Kept it templated so we don't accidentally use the old `sdist` with a new version number in the next release

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 
 package:
   name: dask-core
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/d/dask/dask-{{ version }}.tar.gz
-  sha256: 50ad63035d6429cfa71d4c1832888a668e4260b1ef11163c795edc0ed177b4b3
+  url: https://files.pythonhosted.org/packages/68/e1/1edb5eeb15f88267075c37ee8c83c4527e462bdea328566c5b1e88db705c/dask-{{ version }}.tar.gz
+  sha256: a3972489c46456e54070fe140a2178e002c2e96912ed9ef85c3896b74f469194
```
Comment (jakirkham) [inline recipe/meta.yaml:10]: Filed issue ( https://github.com/conda-forge/dask-core-feedstock/issues/135 ) to revert back to the `pypi.io` URLs

**Comment by jrbourbeau** [comment]
Thanks @jakirkham!

**Comment by jakirkham** [comment]
Happy to help 🙂

Think we are down to the Docker image update 🐳

**Pull request merged**